### PR TITLE
fix escaping

### PR DESCRIPTION
**Pull request by sonnyp**

### CODE REVIEW

File: lib/escape.js [hunk 1]
```diff
@@ -23,5 +23,5 @@ exports.unescapeXML = function unescapeXML (s) {
     .replace(/\&(gt|#62);/g, '>')
     .replace(/\&(quot|#34);/g, '"')
     .replace(/\&(apos|#39);/g, "'")
-    .replace(/\&(nbsp|#160);/g, '\n')
+    .replace(/\&(#xA|#10);/g, '\n')
```
Comment (sonnyp) [inline lib/escape.js]: I don't understand why we need this at all actually

File: lib/escape.js [hunk 2]
```diff
@@ -6,7 +6,16 @@ exports.escapeXML = function escapeXML (s) {
     .replace(/</g, '&lt;')
     .replace(/>/g, '&gt;')
     .replace(/"/g, '&quot;')
-    .replace(/"/g, '&apos;')
+    .replace(/'/g, '&apos;')
```
Comment (sonnyp) [inline lib/escape.js]: was broken :/

File: lib/escape.js [hunk 3]
```diff
@@ -6,7 +6,16 @@ exports.escapeXML = function escapeXML (s) {
...
+}
+
+exports.unescapeXML = function unescapeXML (s) {
+  return s
+    .replace(/\&(amp|#38);/g, '&')
+    .replace(/\&(lt|#60);/g, '<')
+    .replace(/\&(gt|#62);/g, '>')
+    .replace(/\&(quot|#34);/g, '"')
+    .replace(/\&(apos|#39);/g, "'")
```
Comment (sonnyp) [inline lib/escape.js]: just moved unescapeXML function up

@lloydwatkin why do we need to match # codes ?

File: lib/escape.js [hunk 4]
```diff
@@ -16,12 +25,9 @@ exports.escapeXMLText = function escapeXMLText (s) {
...
 
-exports.unescapeXML = function unescapeXML (s) {
+exports.unescapeXMLText = function escapeXMLText (s) {
   return s
     .replace(/\&(amp|#38);/g, '&')
     .replace(/\&(lt|#60);/g, '<')
     .replace(/\&(gt|#62);/g, '>')
-    .replace(/\&(quot|#34);/g, '"')
-    .replace(/\&(apos|#39);/g, "'")
-    .replace(/\&(nbsp|#160);/g, '\n')
```
Comment (sonnyp) [inline lib/escape.js]: removed that last one not sure to understand why we have to unescape that 

@lloydwatkin ?

**Comment by sonnyp** [comment]
@lloydwatkin ping

File: lib/escape.js (hunk 2 above)
Comment (sonnyp) [inline lib/escape.js]: also what about ```

**Comment by dimatill** [comment]
Guys, need fix also for https://github.com/node-xmpp/ltx/blob/master/lib/parsers/ltx.js#L171

**Pull request merged**